packages/sd-jwt/src/SdJwtOptions.ts

```diff
@@ -0,0 +1,42 @@
+import type { HashName, JwaSignatureAlgorithm } from '@aries-framework/core'
+import type { DisclosureFrame } from 'jwt-sd'
+
+export type SdJwtCreateOptions<Payload extends Record<string, unknown> = Record<string, unknown>> = {
+  holderDidUrl: string
+  issuerDidUrl: string
+  issuerOverrideJsonWebAlgorithm?: JwaSignatureAlgorithm
+  disclosureFrame?: DisclosureFrame<Payload>
+  hashingAlgorithm?: HashName
+}
+
+export type SdJwtReceiveOptions = {
+  issuerDidUrl: string
+  holderDidUrl: string
+}
+
+/**
+ * `includedDisclosureIndices` is not the best API, but it is the best alternative until something like `PEX` is supported
+ */
+export type SdJwtPresentOptions = {
+  holderOverrideJsonWebAlgorithm?: JwaSignatureAlgorithm
+  includedDisclosureIndices?: Array<number>
+
+  /**
+   * This information is received out-of-band from the verifier.
+   * The claims will be used to create a normal JWT, used for key binding.
+   */
+  verifierMetadata: {
+    verifierDid: string
+    nonce: string
+    issuedAt: number
+  }
+}
+
+/**
+ * `requiredClaimKeys` is not the best API, but it is the best alternative until something like `PEX` is supported
+ */
+export type SdJwtVerifyOptions = {
+  holderDidUrl: string
+  verifierDid: string
```

Comment:
The verifierDid is included in the JWT challenge. SO the verifier sends some metadata to the holder, which they then sign with their `cnf` claim key. The verifier needs to check whether the `verifierDid` was actually the same.

Reply:
> SO the verifier sends some metadata to the holder,

Is it 'some metadata' or is it specifically the verifiedDid? Because otherwise it may be more compatible for future api changes to make it more generic (such as `verifierChallenge`). A challenge can be anything.